pkg/doc/did/doc.go

```diff
@@ -328,6 +330,45 @@ const (
 var schemaLoaderV1 = gojsonschema.NewStringLoader(schemaV1)     //nolint:gochecknoglobals
 var schemaLoaderV011 = gojsonschema.NewStringLoader(schemaV011) //nolint:gochecknoglobals
 
+// DID is parsed according to the generic syntax: https://w3c.github.io/did-core/#generic-did-syntax
+type DID struct {
+	Scheme           string // Scheme is always "did"
+	Method           string // Method is the specific DID methods
+	MethodSpecificID string // MethodSpecificID is the unique ID computed or assigned by the DID method
+}
+
+// String returns a string representation of this DID
+func (d *DID) String() string {
+	return fmt.Sprintf("%s:%s:%s", d.Scheme, d.Method, d.MethodSpecificID)
+}
+
+// Parse parses the string according to the generic DID syntax.
+// See https://w3c.github.io/did-core/#generic-did-syntax.
+func Parse(did string) (*DID, error) {
```

Comment:
Perhaps this should be something like `ParseDID` since it is in a package with multiple parsing-like functions?

Reply:
@troyronda there's one other parse function - `did.ParseDocument()`. 

To me, `did.Parse()` makes perfect sense as a function that parses a DID.

Let's see what the others think. @sudeshrshetty @rolsonquadras @soluchok @fqutishat